clients/src/main/java/org/apache/kafka/clients/consumer/internals/RequestManager.java

```diff
@@ -18,11 +18,18 @@
 
 import org.apache.kafka.clients.consumer.internals.NetworkClientDelegate.PollResult;
 
+import java.io.Closeable;
+
 /**
  * {@code PollResult} consist of {@code UnsentRequest} if there are requests to send; otherwise, return the time till
  * the next poll event.
  */
-public interface RequestManager {
+public interface RequestManager extends Closeable {
```

Comment:
It seems that none of the request managers implements `close`. Does this need to be `Closeable`?

Reply:
The `FetchRequestManager` does by virtue of extending from `AbstractFetch`. The sole reason for that it is so that it can prepare to send requests to the brokers to close their fetch sessions. It's kind of kludgey the way it's done, so I'll take another look to see if it can be done more cleanly.